addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_11/A5_11_04.java

```diff
@@ -150,10 +151,10 @@ protected State getPowerMeasurementData() {
                 case WATT:
                     factor = 1;
                     break;
-                case KILOWATT:
-                    factor *= 1000;
                 case MEGAWATT:
                     factor *= 1000;
+                case KILOWATT:
```

Comment:
Just a question, not necessarily needs changing: This fall-through is not obvious from the first look. Why not multiply with 1E6 for MW and break?

Or even better: Use a variable for unit, set that to `MetricPrefix.MEGA(SmarthomeUnit.WATT)` (or MILLI or KILO) and use that instead of `SmarthomeUnit.WATT` l. 165?

Reply:
@dominikkv Would be nice if you consider this for a follow up PR :) Thanks